### PR TITLE
Require SSO authentication to access the routes

### PR DESCRIPTION
**Pull request by cquiroz**
In this PR we require authentication for some of the routes, a token is passed with the POST requests and only valid users can access those routes

### CODE REVIEW

File: modules/model/shared/src/test/scala/observe/model/ModelSuite.scala [hunk 1]
```diff
@@ -7,6 +7,7 @@ import cats.kernel.laws.discipline.*
 import lucuma.core.util.arb.ArbEnumerated.given
 import observe.model.GmosParameters.*
 import observe.model.ObserveModelArbitraries.given
+import observe.model.ObserveModelArbitraries.given_Arbitrary_Operator
```
Comment (rpiaggio) [inline modules/model/shared/src/test/scala/observe/model/ModelSuite.scala:10]: Isn't this import included in the previous one?

File: modules/web/server/src/main/resources/app.conf [hunk 2]
```diff
@@ -46,7 +49,7 @@ web-server {
 # Configuration of the observe engine
 observe-engine {
     # host for the odb
-    odb = "ws://localhost:8082/ws"
+    odb = "wss://lucuma-postgres-odb-dev.herokuapp.com/ws"
```
Comment (rpiaggio) [inline modules/web/server/src/main/resources/app.conf:52]: Can we also allow an override for this so we can use a local ODB? I guess an env variable will do.

File: modules/web/server/src/main/scala/observe/web/server/http4s/WebServerLauncher.scala [hunk 3]
```diff
@@ -101,10 +105,30 @@ object WebServerLauncher extends IOApp with LogInitialization {
...
+        .initial(
+          serviceJwt = sso.serviceToken,
+          ssoRoot = sso.ssoUrl,
+          jwtReader = jwtReader[F](sso),
+          httpClient = httpClient
+        )
+        .map(_.map(_.user))
+    )
+
+  // SsoJwtReader(JwtDecoder.withPublicKey(sso.publicKey)).map { jwtReader =>
```
Comment (rpiaggio) [inline modules/web/server/src/main/scala/observe/web/server/http4s/WebServerLauncher.scala:126]: Commented code

**Review by rpiaggio — APPROVED** [review]
LGTM

File: modules/model/shared/src/test/scala/observe/model/ModelSuite.scala (hunk 1 above)
Comment (cquiroz) [reply modules/model/shared/src/test/scala/observe/model/ModelSuite.scala:10]: Ups

File: modules/web/server/src/main/resources/app.conf (hunk 2 above)
Comment (cquiroz) [reply modules/web/server/src/main/resources/app.conf:52]: I wonder how. with an env variable?

**Pull request merged**